sv.c

```diff
@@ -4782,6 +4783,448 @@ Perl_sv_setsv_flags(pTHX_ SV *dsv, SV* ssv, const I32 flags)
+    }
+    assert(SvTYPE(dsv) == SVt_PVIV || SvTYPE(dsv) == SVt_PVNV || SvTYPE(dsv) == SVt_PVMG);
+
+    /* [ 92.5% of calls to this function made it here. ] */
+
+    /* This is the only place we set dsv's flags, with the exception
+       of SVf_IsCOW, which can't be on for S_newSVsv_flags_NN_POK */
+    SvFLAGS(dsv) |= sflags & (
+            SVf_IOK|SVp_IOK|SVf_IVisUV|SVf_NOK|SVp_NOK
+            |SVf_ROK|SVf_POK|SVp_POK|SVf_UTF8|SVppv_STATIC
+    );
+
+            SvCUR_set(dsv, 0);
+            SvLEN_set(dsv, 0);
+            SvPV_set(dsv, NULL);
+
+    switch(sflags & (SVp_IOK|SVp_NOK|SVf_ROK|SVp_POK|SVf_FAKE|SVppv_STATIC)) {
+        case SVp_IOK:  /* [ 50% ]*/
+            SvIV_set(dsv, SvIVX(ssv));
+            SvCUR_set(dsv, 0);
+            SvLEN_set(dsv, 0);
+            SvPV_set(dsv, NULL);
+            return dsv;
+        case SVp_POK|SVp_IOK|SVp_NOK:  /* [ 3% ] */
+            ASSUME(SvTYPE(dsv) != SVt_PVIV);
+            SvNV_set(dsv, SvNVX(ssv));
+            /* FALLTHROUGH */
+        case SVp_POK|SVp_IOK:  /* [ 7% ] */
+            SvIV_set(dsv, SvIVX(ssv));
+            /* FALLTHROUGH */
+        case SVp_POK:  /* [ 28% ] */
+            break;
+        case SVp_POK|SVp_IOK|SVp_NOK|SVppv_STATIC:  /* [ 6.5% ]*/
+            /* e.g. PL_sv_yes, PL_sv_no */
+            ASSUME(!(SvFLAGS(dsv) & SVf_BREAK));
+            ASSUME(SvTYPE(dsv) >= SVt_PVNV);
+            SvFLAGS(dsv) |= SVf_IsCOW;
+
+            SvPV_set(dsv, SvPVX(ssv));
+            SvLEN_set(dsv, 0);
+            SvCUR_set(dsv, SvCUR(ssv));
+
+            SvIV_set(dsv, SvIVX(ssv));
+            SvNV_set(dsv, SvNVX(ssv));
+            return dsv;
+        case SVp_POK|SVp_NOK:  /* [ 3% ]*/
+            SvNV_set(dsv, SvNVX(ssv));
+            break;
+        case SVf_ROK:  /* [ 3% ]*/
+            SvRV_set(dsv, SvREFCNT_inc(SvRV(ssv)));
+            SvCUR_set(dsv, 0);
+            SvLEN_set(dsv, 0);
+            return dsv;
+        default:  /* [ 2% ]*/
+            SvCUR_set(dsv, 0);
+            SvLEN_set(dsv, 0);
+            SvPV_set(dsv, NULL);
+
+            if(!SvOK(ssv))  /* [ ~2% ]*/
+                return dsv;
+            /* Some cases seem so rare that we may as well let
+             * sv_setsv_flags deal with them. For example:
+             *     SVp_IOK|SVp_NOK
+             *
+             * Some cases are (currently) not naturally occurring:
+             *     SVp_POK|SVppv_STATIC
+             *     SVp_POK|SVp_IOK|SVppv_STATIC
+             *     SVp_POK|SVp_NOK|SVppv_STATIC
+             *
+             * Other cases are also rare but also trickier to handle,
+             * so keeps this function smaller to not even try. */
+            sv_setsv_flags(dsv,ssv,flags);
+            return dsv;
+        case SVp_NOK:  /* [ << 1% ]*/
+            ASSUME(SvTYPE(dsv) != SVt_PVIV);
+            SvNV_set(dsv, SvNVX(ssv));
+            SvCUR_set(dsv, 0);
+            SvLEN_set(dsv, 0);
+            SvPV_set(dsv, NULL);
+            return dsv;
```

Comment:
```
        case SVf_ROK:  /* [ 3% ]*/
            SvRV_set(dsv, SvREFCNT_inc(SvRV(ssv)));
            SvCUR_set(dsv, 0);
            SvLEN_set(dsv, 0);
            return dsv;
```
```
            SvCUR_set(dsv, 0);
            SvLEN_set(dsv, 0);
            SvPV_set(dsv, NULL);
            return dsv;
``` 
make these 2 tails to be in identical assignment order so a CC coulde dedupe them, bottom  quote sets slot 4 last, top quote sets slot 4 first.

Reply:
CUR and LEN should now be assigned immediately prior to the return in all cases.